### PR TITLE
Master purchase improve onboarding

### PR DESCRIPTION
**Pull request by hdh-odoo**
**Description of the issue/feature this PR addresses:**

Task:https://www.odoo.com/web#id=2029380&action=327&model=project.task&view_type=form&menu_id=4720
Pad:https://pad.odoo.com/p/r.033eca3b9e0fa87a4c631c82ba21fcf9

**Description**
Made tree view of vendors editable and added dropdown to select fields.
Modified section name in purchase tab on product form.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/purchase/views/product_views.xml [hunk 1]
```diff
@@ -13,10 +13,40 @@
...
+                                <field name="date_end" optional="hide"/>
+                                <field name="product_id" groups="product.group_product_variant" domain="[('product_tmpl_id', '=', product_tmpl_id)]" options="{'no_create_edit': True}" optional="hide"/>
+                                <field name="product_name" optional="hide"/>
+                                <field name="product_code" optional="hide"/>
+                                <field name="delay" string="Lead Time" optional="hide"/>
+                                <field name="company_id" options="{'no_create': True}" optional="hide"/>
+                                <field name="min_qty" optional="hide"/>
+                                <field name="price" optional="show"/>
+                            </tree>
+                        </field>
```
Comment (svs-odoo) [inline addons/purchase/views/product_views.xml]: As you have two time the exact same arch, you can create a record for your list and then pass its `id` in the field  using the `tree_view_ref` in the context.

File: addons/purchase_requisition/views/product_views.xml [hunk 2]
```diff
@@ -35,6 +35,12 @@
                     <field name="purchase_requisition" widget="radio"/>
                 </group>
             </group>
+            <xpath expr="//field[@name='seller_ids']/tree//field[@name='product_id']" position="after">
+                <field name="purchase_requisition_id" attrs="{'invisible': [('purchase_requisition_id', '=', False)]}" optional="hide"/>
+            </xpath>
+            <xpath expr="//field[@name='variant_seller_ids']/tree//field[@name='product_id']" position="after">
+                <field name="purchase_requisition_id" attrs="{'invisible': [('purchase_requisition_id', '=', False)]}" optional="hide"/>
+            </xpath>
```
Comment (svs-odoo) [inline addons/purchase_requisition/views/product_views.xml]: When you will apply my first comment, you will need only one xpath, so don't forget to modify this too.

Comment (svs-odoo) [reply addons/purchase_requisition/views/product_views.xml]: Also, I'm not sure this conditional invisible attribute makes any sense in a list view.
What you try to achieve here ?

File: odoo/addons/test_main_flows/static/tests/tours/main_flow.js [hunk 3]
```diff
@@ -195,12 +195,10 @@ tour.register('main_flow_tour', {
     position: 'right',
     run: "text 1",
 }, {
-    trigger:  ".modal-footer .btn-primary:first",
-    content: _t('Save & Close'),
-    position: 'bottom',
-}, {
+    trigger: 'label:contains("Vendor Taxes")',
+    // click somewhere else to exit cell focus
+},{
```
Comment (svs-odoo) [inline odoo/addons/test_main_flows/static/tests/tours/main_flow.js]: Don't make changes in test files in independent commit. If you need to make adaptation in a test, do it in the same commit who introduces these changes.
The purpose to this is to have each commit able to pass the tests without need to wait further commit.

File: addons/purchase/views/product_views.xml [hunk 4]
```diff
@@ -13,10 +13,12 @@
                 </xpath>
                 <group name="purchase" position="before">
                     <group string="Vendor Prices" name="vendors" attrs="{'invisible': [('product_variant_count','&gt;',1)]}">
-                        <field name="seller_ids" context="{'default_product_tmpl_id':context.get('product_tmpl_id',active_id), 'product_template_invisible_variant': True}" nolabel="1"/>
+                        <field name="seller_ids" context="{'default_product_tmpl_id':context.get('product_tmpl_id',active_id), 'product_template_invisible_variant': True, 'tree_view_ref':'product.product_supplierinfo_tree_view2'}" nolabel="1">
+                        </field>
```
Comment (svs-odoo) [inline addons/purchase/views/product_views.xml]: Keep the tag closed:
``` xml
<field name="seller_ids" context="{'default_product_tmpl_id':context.get('product_tmpl_id',active_id), 'product_template_invisible_variant': True, 'tree_view_ref':'product.product_supplierinfo_tree_view2'}" nolabel="1"/>
```
Same for `variant_seller_ids`.

**Comment by hdh-odoo** [comment]
@svs-odoo I have made the changes. Thanks.

**Comment by sle-odoo** [comment]
robodoo r+

**Comment by sle-odoo** [comment]
robodoo r+
